### PR TITLE
Add ability to skip TLS verification for amtool

### PR DESCRIPTION
**Pull request by nedvna**
Just like with prometheus or alertmanager endpoints, we sometimes need to disable TLS certificate check. We can do that providing `<tls_config>` section there.  
amtool is a simpler thing but it does provide us some basic features like providing custom port and performing basic auth. I think skipping tls verification will fit in here too.

### CODE REVIEW

File: cli/root.go [hunk 1]
```diff
@@ -98,6 +108,7 @@ func Execute() {
 	app.Flag("alertmanager.url", "Alertmanager to talk to").URLVar(&alertmanagerURL)
 	app.Flag("output", "Output formatter (simple, extended, json)").Short('o').Default("simple").EnumVar(&output, "simple", "extended", "json")
 	app.Flag("timeout", "Timeout for the executed command").Default("30s").DurationVar(&timeout)
+	app.Flag("skip.verify", "Skip TLS certificate verification").BoolVar(&skipVerify)
```
Comment (roidelapluie) [inline cli/root.go:111]:
```suggestion
	app.Flag("tls.insecure.skip.verify", "Skip TLS certificate verification").BoolVar(&skipVerify)
```

To match go tls name.

File: cli/root.go [hunk 2]
```diff
@@ -78,6 +81,13 @@ func NewAlertmanagerClient(amURL *url.URL) *client.Alertmanager {
 
 	cr := clientruntime.New(address, path.Join(amURL.Path, defaultAmApiv2path), schemes)
 
+	if amURL.Scheme == "https" && skipVerify {
```
Comment (roidelapluie) [inline cli/root.go:84]: Now I notice this. We should not check the scheme, but apply this all the time. You could go to an HTTP endpoint that redirects you (HTTP 302) to HTTPS.

Comment (nedvna) [reply cli/root.go]: Didn't think we automatically followed them.  
Thanks, removed it.

File: cli/root.go [hunk 3]
```diff
@@ -151,5 +162,9 @@ static configuration:
 
 	date.format
 		Sets the output format for dates. Defaults to "2006-01-02 15:04:05 MST"
+
+	skip.verify
```
Comment (roidelapluie) [inline cli/root.go:166]: This was not updated

File: README.md [hunk 4]
```diff
@@ -315,6 +315,9 @@ output: extended
 
 # Set a default receiver
 receiver: team-X-pager
+
+# Skip TLS certificate verification
+tls.insecure.skip.verify: true
```
Comment (roidelapluie) [inline README.md:320]: Let's get this out of the example configuration to keep it simple and not promote insecure settings.

File: cli/root.go [hunk 5]
```diff
@@ -35,6 +37,7 @@ var (
 	alertmanagerURL *url.URL
 	output          string
 	timeout         time.Duration
+	skipVerify      bool
```
Comment (roidelapluie) [inline cli/root.go:40]: Can we have tlsInsecureSkipVerify ?

**Review by roidelapluie — COMMENTED** [review]
Sorry, I noticed one thing that was not updated, so I added comments from 2 other small things :) Then it will be OK :)

**Comment by nedvna** [comment]
> Sorry, I noticed one thing that was not updated, so I added comments from 2 other small things :) Then it will be OK :)

No problem.  
No I hope I'm not breaking any more naming conventions :)

**Review by roidelapluie — APPROVED** [review]

**Comment by roidelapluie** [comment]
Thanks!

**Pull request merged**